src/mono/mono/metadata/class-setup-vtable.c

```diff
@@ -773,6 +773,13 @@ mono_method_get_method_definition (MonoMethod *method)
 static gboolean
 verify_class_overrides (MonoClass *klass, MonoMethod **overrides, int onum)
 {
+	// on windows and arm, we define NDEBUG for release builds
+	// on browser and wasi, we define DEBUG for debug builds
+#if defined(NDEBUG) || !defined(DEBUG)
```

Comment:
in src/mono/CMakeLists.txt, we define `ENABLE_CHECKED_BUILD` when the `CMAKE_BUILD_TYPE` is `"Debug"`.  Maybe use that? we use it for other kind of slow checks.